app/serializers/search_and_compare/course_serializer.rb

```diff
@@ -110,6 +114,38 @@ def fees
       end
     end
 
+    def contact_details
+      external_contact_info = object.provider.external_contact_info
+
+      {
+        **default_contact_details_value,
+        Phone: external_contact_info['telephone'],
+        Email: external_contact_info['email'],
+        Website: external_contact_info['website'],
+        Address: provider_full_address
+      }
+    end
+
+    def default_contact_details_value
+      {
+        Id: 0,
+        Course: nil,
+        Fax: nil,
+      }
+    end
+
+    def provider_full_address
+      external_contact_info = object.provider.external_contact_info
```

Comment:
I think that the results of `external_contact_info` isn't cached:

```
[2] pry(main)> p.external_contact_info
  ProviderEnrichment Load (17.8ms)  SELECT  "provider_enrichment".* FROM "provider_enrichment" WHERE "provider_enrichment"."provider_code" = $1 AND "provider_enrichment"."status" = $2 ORDER BY "provider_enrichment"."last_published_at" DESC LIMIT $3  [["provider_code", "1Z3"], ["status", 1], ["LIMIT", 1]]
=> {"address1"=>"Eastcote Lane", "address2"=>"South Harrow", "address3"=>"Harrow", "address4"=>"", "postcode"=>"HA2 9AG", "region_code"=>"london", "telephone"=>"02084224503", "email"=>"rsharma.310@lgflmail.org", "website"=>""}
[3] pry(main)> p.external_contact_info
  ProviderEnrichment Load (0.5ms)  SELECT  "provider_enrichment".* FROM "provider_enrichment" WHERE "provider_enrichment"."provider_code" = $1 AND "provider_enrichment"."status" = $2 ORDER BY "provider_enrichment"."last_published_at" DESC LIMIT $3  [["provider_code", "1Z3"], ["status", 1], ["LIMIT", 1]]
=> {"address1"=>"Eastcote Lane", "address2"=>"South Harrow", "address3"=>"Harrow", "address4"=>"", "postcode"=>"HA2 9AG", "region_code"=>"london", "telephone"=>"02084224503", "email"=>"rsharma.310@lgflmail.org", "website"=>""}
```

Can we pass this value through to `provider_full_address`? Or maybe create a method that memoises this value?